### ISSUE TITLE
Problem with validateIncludedElement function and belongsToMany

### ISSUE THREAD

**Issue by davidbnk** [sequelize/sequelize#6491]
https://github.com/sequelize/sequelize/blob/master/lib/model.js#L490

`through.tableName` is undefined in some cases

Code to reproduce:

```
var Sequelize = require('Sequelize');

var sequelize = new Sequelize('sqlite://test.sqlite', {
    dialect: 'sqlite',
    storage: ':memory:'
});

var User = sequelize.define('User', {
    id: {
        type: Sequelize.INTEGER,
        primaryKey: true
    }
});

var Project = sequelize.define('Project', {
    id: {
        type: Sequelize.INTEGER,
        primaryKey: true
    }
});

var UserProjects = sequelize.define('UserProjects', {
    status: Sequelize.STRING
});

User.belongsToMany(Project, { through: UserProjects });
Project.belongsToMany(User, { through: UserProjects });

sequelize.sync().then(function() {

    return Project.findAll({
        include : [{
            model : User,
            where : {
                id : 1
            }
        }],
        order: [
            Sequelize.fn('RANDOM')
        ]
    }).then(function(rows) {
        console.log(rows);
    });
});
```

During the findAll, if you `console.log(tableNames);` after the if (https://github.com/sequelize/sequelize/blob/master/lib/model.js#L492) will get the following output:

```
{ Projects: true, Users: true, undefined: true }
```

**Comment by felixfbecker** [comment]
I don't understand your question. Could you rephrase it?

**Comment by davidbnk** [comment]
@felixfbecker variable `tableNames` in the function `validateIncludedElement` is assigned an undefined key in this line: https://github.com/sequelize/sequelize/blob/master/lib/model.js#L490 because `through.tableName` is undefined, the result of  `tableNames`will be `{ undefined: true }`

**Comment by felixfbecker** [comment]
Okay, so that's the internals. How does this bug surface in your app?

**Comment by davidbnk** [comment]
I was checking databases queries when I found a query like this: `PRAGMA table_info('undefined')`
Sequelize does it before running the actual query
